velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -36,4 +38,101 @@ struct YearFunction : public InitSessionTimezone<T> {
     result = getYear(getDateTime(date));
   }
 };
+
+template <typename T>
+struct UnixTimestampFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  // unix_timestamp();
+  // If no parameters, return the current unix timestamp.
+
+  FOLLY_ALWAYS_INLINE void call(int64_t& result) {
+    result = Timestamp::now().getSeconds();
+  }
+
+  // unix_timestamp(input);
+  // If format is not specified, assume kDefaultFormat.
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const core::QueryConfig& config,
+      const arg_type<Varchar>* /*input*/) {
+    format_ = buildJodaDateTimeFormatter(kDefaultFormat_);
+    isConstFormat_ = true;
+    setTimezone(config);
+  }
+
+  FOLLY_ALWAYS_INLINE bool call(
+      int64_t& result,
+      const arg_type<Varchar>& input) {
+    DateTimeResult dateTimeResult;
+    try {
+      dateTimeResult =
+          format_->parse(std::string_view(input.data(), input.size()));
+    } catch (const VeloxUserError&) {
+      // Return null if could not parse.
```

Comment:
Is this matching SparkSQL semantics? Would you add a link to documentation or explain how you check that in the PR description?

Reply:
Will do. Spark returns null if the format could not be parsed.